src/components/Calculator.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
+import buttonsData from './buttonsData';
+
+function Calculator() {
+  const [calcData, setcalcData] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
+  const handleClick = (btn) => {
+    const data = calculate(calcData, btn);
+    setcalcData(data);
+  };
+  const { total, next, operation } = calcData;
+  return (
+    <div className="container">
+      <div className="screen">
+        {total}
+        {operation}
+        {next }
+      </div>
+      <div className="calculator">
+        <div className="table">
+          {buttonsData.map((button) => (
+            <button
+              key={button.value}
+              type="button"
+              className={button.className}
+              onClick={() => handleClick(button.value)}
+            >
+              {button.label}
+            </button>
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+}
```

Comment:
- [ ] Kindly note that you are required to have at least a child component and pass props to it. Currently you do not have that. The imported `buttonsData` is not a component - it is an array of object. Kindly refactor this component to use at least one child component.

Reply:
@DammyShittu   Oops, I thought I had added all the requirements. I apologize for any confusion or inconvenience this may have caused. I appreciate your feedback. : Happy Coding :smile:

Reply:
@DammyShittu  Oops, I started working on it on an isolated local branch before merging the component branch. I apologize for any confusion or inconvenience this may have caused. I appreciate your feedback. : Happy Coding :smile: